common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/PurchasedProductsFetcher.kt

```diff
@@ -33,61 +33,40 @@ class PurchasedProductsFetcher(
             return
         }
 
-        billing.queryAllPurchases(
+        billing.queryPurchases(
             appUserID,
-            onReceivePurchaseHistory = { allPurchases ->
-                billing.queryPurchases(
-                    appUserID,
-                    onSuccess = { activePurchasesByHashedToken ->
-                        val purchasedProductIdentifiers = allPurchases.map { it.skus[0] }.toSet()
-                        val activeProducts = activePurchasesByHashedToken.values.map { it.skus[0] }.toSet()
-                        val purchasedProducts = purchasedProductIdentifiers.map { productIdentifier ->
-                            createPurchasedProduct(
-                                allPurchases,
-                                productIdentifier,
-                                activeProducts,
-                                productEntitlementMapping
-                            )
-                        }
-                        onSuccess(purchasedProducts)
-                    },
-                    onError
-                )
+            onSuccess = { activePurchasesByHashedToken ->
+                val activePurchases = activePurchasesByHashedToken.values.toList()
+                val purchasedProducts = activePurchases.map {
+                    createPurchasedProduct(it, productEntitlementMapping)
+                }
```

Comment:
Here I'm assuming there can't be more than 1 active purchase with the same product id. I think that's safe to assume but lmk if I'm wrong.